whirlpool/eventsocket.py

```diff
@@ -72,7 +94,17 @@ async def _run(self):
                         LOGGER.debug(
                             f"Stopping receiving. Message type: {str(msg.type)}"
                         )
+
+                        if not self._auth.is_access_token_valid() or msg.data==STATUS_UNAUTHORIZED: 
+                            LOGGER.debug("auth key expired, doing reauth now") 
+                            await self._auth.do_auth()
+
+                        elif msg.data==STATUS_GOING_AWAY: # if going away is issued without and expired key, it appears the server is going down.
```

Comment:
```suggestion
                        elif msg.data==STATUS_GOING_AWAY:
```
No need for comment since it is part of the WS spec

Reply:
I was hoping to preserve the order (elif) with a comment as to why it's in this order... suppose we can put in the bit bucket.
Unfortunately, they are issuing a 1001 when the key expires, and when they are rebooting or whatever else. 
In this configuration, if we get a disconnect and the key is expired, just renew and try again.  If it's not expired, wait longer....